### PR TITLE
CORTX-30169 Added support for K8s Release v0.3.0

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- Added support for K8s Release v0.3.0:  https://github.com/Seagate/cortx-k8s/tree/v0.3.0

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM  - 3N - https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/setup-cortx-cluster/165/ and 1N - https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/setup-cortx-cluster/164/ 

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: solutions/kubernetes/cortx-deploy-functions.sh [hunk 1]
```diff
@@ -79,31 +79,29 @@ function install_yq() {
...
         deployment_method=$DEPLOYMENT_METHOD yq e -i '.solution.deployment_type = env(deployment_method)' solution.yaml
         
         yq e -i '.solution.secrets.name = "cortx-secret"' solution.yaml
-        yq e -i '.solution.secrets.content.kafka_admin_secret = "Seagate@123"' solution.yaml
-        yq e -i '.solution.secrets.content.consul_admin_secret = "Seagate@123"' solution.yaml
-        yq e -i '.solution.secrets.content.common_admin_secret = "Seagate@123"' solution.yaml
+        yq e -i '.solution.secrets.content.kafka_admin_secret = "null"' solution.yaml
+        yq e -i '.solution.secrets.content.consul_admin_secret = "null"' solution.yaml
+        yq e -i '.solution.secrets.content.common_admin_secret = "null"' solution.yaml
         yq e -i '.solution.secrets.content.s3_auth_admin_secret = "ldapadmin"' solution.yaml
```
Comment (gauravchaudhari02) [inline solutions/kubernetes/cortx-deploy-functions.sh:89]: looks like this is also changed to null in v0.3.0

File: solutions/kubernetes/cortx-deploy-functions.sh [hunk 2]
```diff
@@ -79,31 +79,29 @@ function install_yq() {
...
-        yq e -i '.solution.secrets.content.csm_auth_admin_secret = "seagate2"' solution.yaml
+        yq e -i '.solution.secrets.content.csm_auth_admin_secret = "null"' solution.yaml
         yq e -i '.solution.secrets.content.csm_mgmt_admin_secret = "Cortxadmin@123"' solution.yaml
```
Comment (gauravchaudhari02) [inline solutions/kubernetes/cortx-deploy-functions.sh:91]: looks like this is also changed to null in v0.3.0

Comment (shailesh-vaidya) [reply solutions/kubernetes/cortx-deploy-functions.sh:91]: We use `s3_auth_admin_secret` and `csm_mgmt_admin_secret` in automated IO testing hence not changing them to null as of now. 

Also as per Walter - 
`
Default passwords are no longer provided in solution.example.yaml (in solution.secrets.content).  Best practice is either to leave these all blank (and let deploy-cortx-cloud.sh generate passwords), or to specify passwords for the default IAM account (s3_auth_admin_secret) and CSM (csm_mgmt_admin_secret) and let deploy-cortx-cloud.sh generate the rest.
`

File: solutions/kubernetes/cortx-deploy-functions.sh (hunk 1 above)
Comment (shailesh-vaidya) [reply solutions/kubernetes/cortx-deploy-functions.sh:89]: We use `s3_auth_admin_secret` and `csm_mgmt_admin_secret` in automated IO testing hence not changing them to null as of now. 

Also as per Walter - 
`
Default passwords are no longer provided in solution.example.yaml (in solution.secrets.content).  Best practice is either to leave these all blank (and let deploy-cortx-cloud.sh generate passwords), or to specify passwords for the default IAM account (s3_auth_admin_secret) and CSM (csm_mgmt_admin_secret) and let deploy-cortx-cloud.sh generate the rest.
`

**Review by gauravchaudhari02 — DISMISSED** [review]

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**